### PR TITLE
fix: Proprietary models now get correctly shown in leaderboard

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
Proprietary models were not correctly shown due to model size filtering being incorrectly implemented.
I also fixed column wrapping and formatting, the per task performance table is also correctly ordered.

### CODE REVIEW

**Comment by x-tabdeveloping** [comment]
Hmmm tests are failing for a seemingly unrelated reason...

**Comment by isaac-chung** [comment]
Reruns are successful now 👍

**Review by KennethEnevoldsen — APPROVED** [review]

File: mteb/leaderboard/table.py [hunk 1]
```diff
@@ -138,34 +139,43 @@ def scores_to_tables(
...
     # Removing HF organization from model
     joint_table["model_name"] = joint_table["model_name"].map(
         lambda name: name.split("/")[-1]
     )
     # Adding markdown link to model names
-    joint_table["model_name"] = (
-        "[" + joint_table["model_name"] + "](" + joint_table.pop("model_link") + ")"
+    name_w_link = (
+        "[" + joint_table["model_name"] + "](" + joint_table["model_link"] + ")"
     )
```
Comment (KennethEnevoldsen) [inline mteb/leaderboard/table.py:174]: Non-blocking: I would probably just create the column initially as model_name_w_link using a method on the model_meta object like model_meta.model_name_as_markdown(). 

(could imagine we could use this elsewhere and also removed completely from the leaderboard code)

**Review by KennethEnevoldsen — APPROVED** [review]

File: mteb/leaderboard/table.py (hunk 1 above)
Comment (x-tabdeveloping) [reply mteb/leaderboard/table.py:174]: Yeah, seems reasonable, I think I will do this in a future PR

**Pull request merged**